MaxText/layers/models.py

```diff
@@ -184,7 +200,16 @@ def __call__(self,
     y = y.astype(cfg.dtype)
 
     if cfg.use_positional_embedding:
-      y = PositionalEmbedding(cfg.base_emb_dim)(y, decoder_positions)
+      if cfg.trainable_position_size:
```

Comment:
Look, we've kind of combined two separate ideas. 

(1) use_positional_embedding which controls a positional embedding
(2) tranable_position_size which if greater than 0 adds a second embedding strategy.

But they shouldn't be tied together.

Reply:
Hi, @rwitten, here's some info I should add in the comments

`use_positional_embedding` is to control whether to add a positional embedding.
And positional_embedding can be a trainable one with a positive `tranable_position_size` or [an untrainable one](https://github.com/google/maxtext/blob/ec9cdce1928fb36f93e498c8f74f8f130bad3dc2/MaxText/layers/embeddings.py#L187).

```
    if cfg.use_positional_embedding:
      if cfg.trainable_position_size:
        # trainable positional embedding, same implementation as `Embed`
        y += Embed(
          num_embeddings=cfg.trainable_position_size,
          features=cfg.emb_dim,
          dtype=cfg.dtype,
          embedding_init=nn.initializers.normal(stddev=1.0),
          name='position_embedder',
          config=cfg)(decoder_positions)
      else:
        # untrainable positional embedding
        y = PositionalEmbedding(cfg.base_emb_dim)(y, decoder_positions)
```

Reply:
There are three possible outcomes. Hence a binary won't suffice but with two flags we can do it. Additionally, someone could do both! I would much prefer:

```
if cfg.use_untrainable_positional_embedding:
  y = PositionalEmbedding(cfg.base_emb_dim)(y, decoder_positions)
  
if cfg.trainable_position_size >= 0:
          y += Embed(
          num_embeddings=cfg.trainable_position_size,
          features=cfg.emb_dim,
          dtype=cfg.dtype,
          embedding_init=nn.initializers.normal(stddev=1.0),
          name='position_embedder',
          config=cfg)(decoder_positions)
 
```

Reply:
Will follow your advice.